internal/operators/ocs/validations.go

```diff
@@ -0,0 +1,133 @@
+package ocs
+
+import (
+	"encoding/json"
+	"math"
+
+	"github.com/docker/go-units"
+	"github.com/openshift/assisted-service/internal/host"
+	"github.com/openshift/assisted-service/models"
+	"github.com/sirupsen/logrus"
+)
+
+//go:generate mockgen -source=validations.go -package=ocs -destination=mock_validations.go
+type OcsValidator interface {
+	ValidateOCSRequirements(cluster *models.Cluster) bool
+}
+
+type ocsValidator struct {
+	*Config
+	log     logrus.FieldLogger
+	hostApi host.API
+}
+
+func NewOcsValidator(log logrus.FieldLogger, hostApi host.API, cfg *Config) OcsValidator {
+	return &ocsValidator{
+		log:     log,
+		Config:  cfg,
+		hostApi: hostApi,
+	}
+}
+
+type Config struct {
+	OCSMinimumCPUCount        int64 `envconfig:"OCS_MINIMUM_CPU_COUNT" default:"24"`
+	OCSMinimumRAMGB           int64 `envconfig:"OCS_MINIMUM_RAM_GB" default:"66"`
+	OCSRequiredDisk           int64 `envconfig:"OCS_MINIMUM_DISK" default:"3"`
+	OCSRequiredHosts          int   `envconfig:"OCS_MINIMUM_HOST" default:"3"`
+	OCSRequiredCPUCount       int64 `envconfig:"OCS_REQUIRED_CPU_COUNT" default:"30"`
+	OCSRequiredRAMGB          int64 `envconfig:"OCS_REQUIRED_RAM_GB" default:"72"`
+	OCSRequiredMasterCPUCount int64 `envconfig:"OCS_REQUIRED_MASTER_CPU_COUNT" default:"16"`
+	OCSRequiredMasterRAMGB    int64 `envconfig:"OCS_REQUIRED_MASTER_RAM_GB" default:"64"`
+	OCSMasterWorkerHosts      int   `envconfig:"OCS_REQUIRED_MASTER_WORKER_HOSTS" default:"5"`
+	OCSMinimalDeployment      bool  `envconfig:"OCS_MINIMAL_DEPLOYMENT" default:"false"`
+	OCSStorageCapacity        int64 `envconfig:"OCS_STORAGE_CAPACITY" default:"10"` // to claim for OSDs
+	OCSDisksAvailable         int64 `envconfig:"OCS_DISKS_AVAILABLE" default:"3"`
+}
+
+// ValidateOCSRequirements is used to validate min requirements of OCS
+func (o *ocsValidator) ValidateOCSRequirements(cluster *models.Cluster) bool {
+	hosts := cluster.Hosts
+	if len(hosts) < o.OCSRequiredHosts {
+		o.log.Info("Insufficient hosts to deploy OCS")
+		return false
+	}
+	var CPUCount int64 = 0                //count the total CPUs on the cluster
+	var TotalRAM int64 = 0                // count the total available RAM on the cluster
+	var DiskCount int64 = 0               // count the total disks on all the hosts
+	var MinDiskSize int64 = math.MaxInt64 // min disk size after boot disk size to claim for the PV created by localVolume
+	var HostsWithDisks int = 0            // to determine total number of hosts with disks. OCS requires atleast 3 hosts with non-bootable disks                           // to claim for the PV created by LocalVolumeSet
+	if len(hosts) == o.OCSRequiredHosts { // for only 3 master hosts, we need to install OCS on the masters
```

Comment:
according to your test document - we can install OCS if we have 3 hosts - doesn't matter if it's master or worker so I think your comment is confusing. If we must have 3 masters you should check the host role.